internal/aws/containerinsight/httpclient/httpclient.go

```diff
@@ -0,0 +1,117 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpClient
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+const (
+	defaultMaxRetries       = 3
+	defaultTimeout          = 1 * time.Second
+	defaultBackoffRetryBase = 200 * time.Millisecond
+	maxHTTPResponseLength   = 5 * 1024 * 1024 // 5MB
+)
+
+type HTTPClient struct {
+	client doer
+}
+
+type Requester interface {
+	Request(ctx context.Context, path string) ([]byte, error)
+}
+
+type doer interface {
+	Do(request *retryablehttp.Request) (*http.Response, error)
```

Comment:
We do have already a queued-retry logic for exporters, why is this needed?

Reply:
This is used for calling the ECS container agent introspection task endpoint and cluster endpoint, and there are some specific setting for the response data, I think this is different from the logic for exporters.

Reply:
@Aneurysm9 you know both, please followup if this is a duplicate logic to ensure we use the exporter helper if possible :)